pkg/etc/checker.go

```diff
@@ -0,0 +1,94 @@
+package etc
+
+import (
+	"errors"
+	"fmt"
+	"os"
+
+	log "github.com/sirupsen/logrus"
+)
+
+// Check checks config values to fail fast in case of any problems
+// that we might have due to invalid config.
+func Check(config Config) (err error) {
+	log.WithFields(log.Fields{
+		"pid": os.Getpid(),
+	}).Debug("Current process")
+
+	log.WithFields(log.Fields{
+		"uid":      os.Getuid(),
+		"gid":      os.Getegid(),
+		"home_dir": os.Getenv("HOME"),
+	}).Debug("Current user")
+
+	if config.Trivy.CacheDir == "" {
+		err = errors.New("trivy cache dir must not be blank")
+		return
+	}
+
+	if config.Trivy.ReportsDir == "" {
+		err = errors.New("trivy reports dir must not be blank")
+		return
+	}
+
+	if err = ensureDirExists(config.Trivy.CacheDir, "trivy cache dir"); err != nil {
+		return
+	}
+
+	if err = ensureDirExists(config.Trivy.ReportsDir, "trivy reports dir"); err != nil {
+		return
+	}
+
+	if config.API.IsTLSEnabled() {
+		if !fileExists(config.API.TLSCertificate) {
+			err = fmt.Errorf("TLS certificate file does not exist: %s", config.API.TLSCertificate)
+			return
+		}
+		if !fileExists(config.API.TLSKey) {
+			err = fmt.Errorf("TLS private key file does not exist: %s", config.API.TLSKey)
+			return
+		}
+	}
+
+	return
+}
+
+func ensureDirExists(path, description string) (err error) {
+	if !dirExists(path) {
+		log.WithField("path", path).Warnf("%s does not exist", description)
+		log.WithField("path", path).Debugf("Creating %s", description)
+		if err = os.MkdirAll(path, 0777); err != nil {
+			err = fmt.Errorf("creating %s: %w", description, err)
+			return
+		}
+	}
+	fi, err := os.Stat(path)
+	if err != nil {
+		return
+	}
+
+	log.WithFields(log.Fields{
+		"mode": fi.Mode().String(),
+	}).Debugf("%s permissions", description)
+	return
+}
+
+// dirExists checks if a dir exists before we
+// try using it to prevent further errors.
+func dirExists(name string) bool {
+	info, err := os.Stat(name)
+	if os.IsNotExist(err) {
+		return false
+	}
+	return info.IsDir()
+}
+
+// fileExists checks if a file exists and is not a directory before we
+// try using it to prevent further errors.
+func fileExists(name string) bool {
+	info, err := os.Stat(name)
+	if os.IsNotExist(err) {
+		return false
+	}
+	return !info.IsDir()
+}
```

Comment:
nit:
```suggestion
func fileExists(name string) bool {
	_, err := os.Stat(filename)
	return err == nil
	}
```
I thought the above is enough. Is there any reason to use `info.IsDir()`? If you want to make sure it it not a directory, `err == nil` looks enough as far as I see the following link.
https://goruncode.com/how-to-check-if-a-file-exists/

Reply:
I wanted to return false if the name arg is a directory. So this method also asserts that the path in the actual file, e.g. a TLS certificate file.

Reply:
Oh, I see.